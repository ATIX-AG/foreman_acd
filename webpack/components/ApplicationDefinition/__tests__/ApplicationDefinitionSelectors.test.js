import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  selectShowAlertModal,
  selectAlertModalText,
  selectAlertModalTitle,
  selectEditMode,
  selectAnsiblePlaybook,
  selectServices,
  selectColumns,
  selectHiddenForemanParameterTypes,
  selectParametersData,
  selectAnsibleVarsAll,
  selectParamEditMode,
} from '../ApplicationDefinitionSelectors';

import { applicationDefinitionConfData1 } from '../__fixtures__/applicationDefinitionConfData1.fixtures';

const stateFactory = obj => ({
  foremanAcd: {
    applicationDefinitionConf: obj,
  },
});

const fixtures = {
  'should return showAlertModal from applicationDefinitionConfData1 fixtures': () =>
    selectShowAlertModal(stateFactory(applicationDefinitionConfData1)),
  'should return alertModalText from applicationDefinitionConfData1 fixtures': () =>
    selectAlertModalText(stateFactory(applicationDefinitionConfData1)),
  'should return alertModalTitle from applicationDefinitionConfData1 fixtures': () =>
    selectAlertModalTitle(stateFactory(applicationDefinitionConfData1)),
  'should return editMode from applicationDefinitionConfData1 fixtures': () =>
    selectEditMode(stateFactory(applicationDefinitionConfData1)),
  'should return ansiblePlaybook from applicationDefinitionConfData1 fixtures': () =>
    selectAnsiblePlaybook(stateFactory(applicationDefinitionConfData1)),
  'should return services from applicationDefinitionConfData1 fixtures': () =>
    selectServices(stateFactory(applicationDefinitionConfData1)),
  'should return parametersData from applicationDefinitionConfData1 fixtures': () =>
    selectParametersData(stateFactory(applicationDefinitionConfData1)),
  'should return columns from applicationDefinitionConfData1 fixtures': () =>
    selectColumns(stateFactory(applicationDefinitionConfData1)),
  'should return hiddenForemanParameterTypes from applicationDefinitionConfData1 fixtures': () =>
    selectHiddenForemanParameterTypes(
      stateFactory(applicationDefinitionConfData1)
    ),
  'should return ansibleVarsAll from applicationDefinitionConfData1 fixtures': () =>
    selectAnsibleVarsAll(stateFactory(applicationDefinitionConfData1)),
  'should return ParamEditMode from applicationDefinitionConfData1 fixtures': () =>
    selectParamEditMode(stateFactory(applicationDefinitionConfData1)),
};

describe('ApplicationDefinitionSelectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
