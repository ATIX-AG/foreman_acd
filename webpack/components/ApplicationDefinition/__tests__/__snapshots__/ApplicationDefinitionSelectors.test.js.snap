// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationDefinitionSelectors should return ParamEditMode from applicationDefinitionConfData1 fixtures 1`] = `undefined`;

exports[`ApplicationDefinitionSelectors should return alertModalText from applicationDefinitionConfData1 fixtures 1`] = `undefined`;

exports[`ApplicationDefinitionSelectors should return alertModalTitle from applicationDefinitionConfData1 fixtures 1`] = `undefined`;

exports[`ApplicationDefinitionSelectors should return ansiblePlaybook from applicationDefinitionConfData1 fixtures 1`] = `
Object {
  "groups": Object {
    "all": Array [
      Object {
        "id": 0,
        "name": "repository",
        "value": "https://github.com/bennojoy/mywebapp.git",
      },
    ],
    "dbservers": Array [
      Object {
        "id": 0,
        "name": "mysqlservice",
        "value": "mysqld",
      },
      Object {
        "id": 1,
        "name": "mysql_port",
        "value": "3306",
      },
      Object {
        "id": 2,
        "name": "dbuser",
        "value": "webapp",
      },
      Object {
        "id": 3,
        "name": "dbname",
        "value": "ANSAP01",
      },
      Object {
        "id": 4,
        "name": "upassword",
        "value": "Bond@007",
      },
      Object {
        "id": 5,
        "name": "masterpassword",
        "value": "MySQL@007",
      },
    ],
    "webservers": Array [
      Object {
        "id": 0,
        "name": "dummy_var",
        "value": "0",
      },
    ],
  },
  "id": 2,
  "name": "LAMP",
}
`;

exports[`ApplicationDefinitionSelectors should return ansibleVarsAll from applicationDefinitionConfData1 fixtures 1`] = `
Array [
  Object {
    "id": 0,
    "name": "repository",
    "value": "https://github.com/bennojoy/mywebapp.git",
  },
]
`;

exports[`ApplicationDefinitionSelectors should return columns from applicationDefinitionConfData1 fixtures 1`] = `
Array [
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "formatters": Array [
        null,
      ],
      "label": "Name",
      "props": Object {
        "index": 0,
        "style": Object {
          "width": "15%",
        },
      },
    },
    "property": "name",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "formatters": Array [
        null,
      ],
      "label": "Description",
      "props": Object {
        "index": 1,
        "style": Object {
          "width": "10%",
        },
      },
    },
    "property": "description",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "formatters": Array [
        null,
      ],
      "label": "Hostgroup",
      "props": Object {
        "index": 2,
        "style": Object {
          "width": "20%",
        },
      },
    },
    "property": "hostgroup",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "formatters": Array [
        null,
      ],
      "label": "Ansible Group",
      "props": Object {
        "index": 3,
        "style": Object {
          "width": "20%",
        },
      },
    },
    "property": "ansibleGroup",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "formatters": Array [
        null,
      ],
      "label": "min count",
      "props": Object {
        "index": 4,
        "style": Object {
          "width": "10%",
        },
      },
    },
    "property": "minCount",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "formatters": Array [
        null,
      ],
      "label": "max count",
      "props": Object {
        "index": 5,
        "style": Object {
          "width": "10%",
        },
      },
    },
    "property": "maxCount",
  },
  Object {
    "cell": Object {
      "formatters": Array [
        null,
      ],
    },
    "header": Object {
      "formatters": Array [
        null,
      ],
      "label": "Actions",
      "props": Object {
        "index": 6,
        "style": Object {
          "width": "15%",
        },
      },
    },
    "property": "actions",
  },
]
`;

exports[`ApplicationDefinitionSelectors should return editMode from applicationDefinitionConfData1 fixtures 1`] = `undefined`;

exports[`ApplicationDefinitionSelectors should return hiddenForemanParameterTypes from applicationDefinitionConfData1 fixtures 1`] = `undefined`;

exports[`ApplicationDefinitionSelectors should return parametersData from applicationDefinitionConfData1 fixtures 1`] = `undefined`;

exports[`ApplicationDefinitionSelectors should return services from applicationDefinitionConfData1 fixtures 1`] = `
Array [
  Object {
    "ansibleGroup": "webservers",
    "ansibleParameters": Array [
      Object {
        "id": 0,
        "name": "dummy_var",
        "value": "0",
      },
    ],
    "description": "",
    "foremanParameters": Array [
      Object {
        "description": "",
        "id": 1,
        "locked": false,
        "name": "CP",
        "type": "computeprofile",
        "value": "1",
      },
      Object {
        "description": "",
        "id": 2,
        "locked": true,
        "name": "LE",
        "type": "lifecycleenv",
        "value": "1",
      },
    ],
    "hostgroup": "1",
    "id": 1,
    "maxCount": "",
    "minCount": "2",
    "name": "web",
  },
  Object {
    "ansibleGroup": "dbservers",
    "ansibleParameters": Array [
      Object {
        "id": 0,
        "name": "mysqlservice",
        "value": "mysqld",
      },
      Object {
        "id": 1,
        "locked": true,
        "name": "mysql_port",
        "value": "3306",
      },
      Object {
        "id": 2,
        "name": "dbuser",
        "value": "webapp",
      },
      Object {
        "id": 3,
        "name": "dbname",
        "value": "ANSAP01",
      },
      Object {
        "id": 4,
        "name": "upassword",
        "value": "Bond@007",
      },
      Object {
        "id": 5,
        "name": "masterpassword",
        "value": "MySQL@007",
      },
    ],
    "description": "",
    "foremanParameters": Array [],
    "hostgroup": "1",
    "id": 2,
    "maxCount": "",
    "minCount": "1",
    "name": "db",
  },
]
`;

exports[`ApplicationDefinitionSelectors should return showAlertModal from applicationDefinitionConfData1 fixtures 1`] = `undefined`;
